### DISCUSSION TITLE
Newbie question regarding battery capacity and iterative simulation

### DISCUSSION THREAD

**Discussion by A-bnk** [pybamm-team/PyBaMM#4471] [Q&A]
Hello Guys!

I came across this fantastic project and would like to use it for my own project. I am currently a university student trying to expand a simulator for smart grids, in which charging stations and electric vehicles are part of the system. At the moment, I am using a linear degradation model for simplicity (and to have a minimal working example), e.g a 10% capacity loss every 500 cycles. I would like to replace this with one of your models to make it much more realistic.

The simulation tool I am using performs the simulation in steps. For each simulation step, the following information is available: cycles, battery age (in months), State of Charge (SoC), capacity, and charge/discharge power. What would you suggest would be a simple and easy-to-implement approach? For me, anything would be an improvement over the current one. I would like to have the battery health after each simulation step, which I can then use as the starting point for the next simulation step, etc.. 

```python
degradation(sim_time_since_start_hrs):
    # stuff...
    # pseudo: something along the lines of this, thats really all i need
    battery_capacity_size_new = battery_capacity_size_original * (1 - (cyclic_degration + calendar_aging)) 

```
In one of your notebooks, you guys described the possibility to simulate iteratively (using the starting_solution feature). With every simulation step, I need to be able to set the state of charge (SoC) that was calculated previously. If I understood that correctly, this is done using the initial_soc feature in the solve() method. One last question: I found your second project ("Liionpack") in your PyBaMM team repo. Do you think it would be better to use this instead of PyBaMM (I know it basically uses PyBaMM under the hood)? I would really appreciate it if someone can guide me on how to make a minimal working example. Many thanks!

**Comment by pauljgasper** [comment]
(this is a shameless plug) PyBaMM is way more computationally expensive / the physics are way more detailed than you need. Check out https://github.com/NREL/BLAST-Lite :)

**Reply by A-bnk** [reply]
Hi! Thanks for sharing this project, it looks really promising.
